beacon_node/client/src/builder.rs

```diff
@@ -336,9 +347,52 @@ where
                     "state_root" => ?state_root,
                 );
 
+                let service = match deposit_snapshot_result {
+                    Ok(Some(deposit_snapshot)) => match Eth1Service::from_deposit_snapshot(
+                        config.eth1,
+                        context.log().clone(),
+                        spec,
+                        deposit_snapshot,
+                    )
+                    .await
+                    {
+                        Ok(service) => {
+                            let deposits_loaded = service.get_deposit_snapshot().deposits;
+                            info!(
+                                context.log(),
+                                "Loaded deposit cache from snapshot";
+                                "deposit count" => deposits_loaded,
+                            );
+                            Some(service)
+                        }
+                        Err(e) => {
+                            warn!(context.log(),
+                                "Unable to load deposit snapshot";
+                                "error" => format!("{:?}", e)
+                            );
+                            None
+                        }
+                    },
+                    Ok(None) => {
+                        warn!(
+                            context.log(),
+                            "Error loading deposit snapshot from remote: received NULL"
```

Comment:
In practice this corresponds to a 404 from the remote BN, which is the most likely response if we're checkpoint syncing from a non-Lighthouse node. I think we could make this warning less scary by re-wording it:

```suggestion
                            "Remote BN doesn't support Lighthouse fast deposit sync";
                            "message" => "falling back to regular deposit sync"
```

Reply:
done!